main.c

```diff
@@ -27,14 +30,44 @@ void simulate(struct Hardware *hardware, char *fileList[], int numFiles) {
 
 	int processCompleted = 0;
 	int currProcess = 0;
-
+	uint32_t instructionCount = 0;
+
+	struct Process *process[numFiles];
+	for(int i=0;i<numFiles;i++) {
+		process[i] = (struct Process *) malloc(sizeof(struct Process));
+		process[i]->pid = i;
+		process[i]->state = 1;
+		process[i]->ldt = (struct SegmentTable *) malloc(sizeof(struct SegmentTable));
+		process[i]->ldt->csBase = 0;
+		process[i]->ldt->csLength = 0;
+		process[i]->ldt->dsBase = 0;
+		process[i]->ldt->dsLength = 0;
```

Comment:
@jaskaran1312, @Meloninga54 how should I initialise ldt?

Reply:
this will be prepaged right so it doesn't matter ill write a quick prepage function usme anyways this will be replaced as soon as the process comes for the first time. -1 rakh sakte tab tak

Reply:
Okay.